src/Prairie/Advanced.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE RankNTypes #-}
+module Prairie.Advanced
+    where
+
+import Data.Functor.Const (Const(..))
+import Data.Functor.Identity (Identity(..))
+import Data.Text (Text)
+import Prairie.Class (Field, getRecordField, recordFieldLabel, tabulateRecordA)
+import qualified Prairie.Class as Simple
+import Prelude hiding (sequenceA, zipWith)
+
+newtype Record f rec = Record
+    { getField :: forall x. Field rec x -> f x }
```

Comment:
If we make this not a record then we can define an HasField instance using SymbolToField so you can dot access this. Maybe. Haven’t tested it out

Reply:
okay, I just tested it, and indeed, it does work:

```haskell
data User = User { name :: String, age :: Int }
  deriving (Show, Eq)

mkRecord ''User

deriving instance Eq (Field User a)
deriving instance Show (Field User a)

exampleUser = User "Alice" 30

newtype Decorated f rec = Decorated (forall a. Field rec a -> f a)

instance (SymbolToField sym rec a) => HasField sym (Decorated f rec) (f a) where
    getField :: (HasField sym (Decorated f rec) (f a)) => Decorated f rec -> f a
    getField (Decorated mk) = mk (symbolToField @sym)

traverseUser :: forall f g m. Applicative m => (forall x. f x -> m (g x)) -> Decorated f User -> m (Decorated g User)
traverseUser morph userF = do
    name' <- morph userF.name
    age' <- morph userF.age
    pure $ Decorated \case
        UserName -> name'
        UserAge -> age'

```